core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -502,6 +522,8 @@ private[spark] class BlockManagerInfo(
 
   def blocks: JHashMap[BlockId, BlockStatus] = _blocks
 
+  def cachedBlocks: mutable.HashSet[BlockId] = _cachedBlocks
```

Comment:
I was just following the same pattern as the `blocks` accessor which exposes a mutable Java `HashMap`. Creating a new set which copies all the references is overkill, since the only place this is used to check if this is `nonEmpty`. I can switch this method out with `hasCachedBlocks` which just returns a `Boolean`